egs/wsj/s5/utils/data/get_utt2dur.sh

```diff
@@ -68,7 +68,14 @@ else
       echo  "$0: wav-to-duration is not on your path"
       exit 1;
     fi
-    if ! wav-to-duration scp:$data/wav.scp ark,t:$data/utt2dur 2>&1 | grep -v 'nonzero return status'; then
+
+    read_entire_file=false
+    if [ `cat $data/wav.scp | sed -n '/sox.*speed/p' | wc -l` -gt 0 ]; then
```

Comment:
On 11/04/16 21:16, Daniel Povey wrote:
>
> In egs/wsj/s5/utils/data/get_utt2dur.sh 
> <https://github.com/kaldi-asr/kaldi/pull/678#discussion_r59273232>:
>
> > @@ -68,7 +68,14 @@ else
> >        echo  "$0: wav-to-duration is not on your path"
> >        exit 1;
> >      fi
> > -    if ! wav-to-duration scp:$data/wav.scp ark,t:$data/utt2dur 2>&1 | grep -v 'nonzero return status'; then
> > +
> > +    read_entire_file=false
> > +    if [ `cat $data/wav.scp | sed -n '/sox.*speed/p' | wc -l` -gt 0 ]; then
>
> I think a grep or grep -E command with output piped to /dev/null would 
> be more elegant-- grep returns true if it matched at least one line.
>

instead of stuffing the output to /dev/null you might want to use -q

        -q, --quiet, --silent
               Quiet; do not write anything to standard output.  Exit 
immediately with zero status if any match is found, even if  an error  was
               detected.  Also see the -s or --no-messages option. (-q 
is specified by POSIX.)

Tony

-- 
Speechmatics is a trading name of Cantab Research Limited
We are hiring: www.speechmatics.com/careers 
<https://www.speechmatics.com/careers>
Dr A J Robinson, Founder, Cantab Research Ltd
Phone direct: 01223 794096, office: 01223 794497
Company reg no GB 05697423, VAT reg no 925606030
51 Canterbury Street, Cambridge, CB4 3QG, UK